hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/key/KeyUtils.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.spi.key;
+
+import com.hedera.hapi.node.base.ContractID;
+import com.hedera.hapi.node.base.Key;
+import com.hedera.hapi.node.base.Key.KeyOneOfType;
+import com.hedera.hapi.node.base.KeyList;
+import com.hedera.hapi.node.base.ThresholdKey;
+import com.hedera.pbj.runtime.io.buffer.Bytes;
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * Utility class for working with keys. This validates if the key is empty and valid.
+ */
+// NOTE: This class is not in the right place. But is needed by several modules.
+// !!!!!!!!!!🔥🔥🔥 It should be moved once we find where to keep it. 🔥🔥🔥!!!!!!!!!!!
+public class KeyUtils {
+    public static final int ED25519_BYTE_LENGTH = 32;
+    private static final byte ODD_PARITY = (byte) 0x03;
+    private static final byte EVEN_PARITY = (byte) 0x02;
+    public static final int ECDSA_SECP256K1_COMPRESSED_KEY_LENGTH = 33;
+    /**
+     * Checks if the given key is empty.
+     * For a KeyList type checks if the list is empty.
+     * For a ThresholdKey type checks if the list is empty.
+     * For an Ed25519 or EcdsaSecp256k1 type checks if there are zero bytes.
+     * TODO: This method need to be updated for other key types.
+     * @param pbjKey the key to check
+     * @return true if the key is empty, false otherwise
+     */
+    public static boolean isEmpty(@NonNull final Key pbjKey) {
```

Comment:
Nit: pbjKey is `@Nullable`, as it does permit null (line 47 checks for it and returns true)

Reply:
Thanks Joseph. Fixed